test/test_jit.py

```diff
@@ -2244,6 +2244,32 @@ def forward(self, input, other=four):
         t = Test()
         self.assertEqual(t(torch.ones(1)), torch.ones(1) + 4)
 
+    def test_union_to_optional(self):
+        def test1(u: Union[int, None]) -> int:
+            if u is not None:
+                return u
+            else:
+                return 0
+        scripted = torch.jit.script(test1)
```

Comment:
Optional:

Use `self.checkScript`

Reply:
I was told `self.checkScript` is the old way in one of my earlier PR's.

Reply:
Hmm, didn't know about that. What's wrong with it and what's the alternative?